### DISCUSSION TITLE
Генерация штрихкода

### DISCUSSION THREAD

**Discussion by vldekov** [totumonline/totum-ru-issues-and-discussions#318] [Как это сделать?] [answered]
Добрый день!
Есть задача получить штрихкод при печати этикетки. В Linux есть команда barcode. Как возможно получить штрихкод?

**Comment by erpic-ru** [comment]
Подключить шрифт. Я повторил [этот фокус](https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/162#discussioncomment-3824833), все отлично работает

**Reply by vldekov** [reply]
Так же возможно сделать и линейный штрихкод ( code39, code128 ) ?

**Reply by erpic-ru** [reply]
<img width="285" alt="Снимок экрана 2023-01-23 в 09 58 50" src="https://user-images.githubusercontent.com/91674114/213982066-d9660175-ac59-4caf-9a8a-9ec2f8bd7322.png"> - вот 128

В печатной форме в стилях подключаете шрифт: 
```

@import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600|Roboto:400,400i,700,700i,500|Roboto+Mono:400,700&amp;subset=cyrillic");
body {font-family: 'Roboto', sans-serif; }

Ещё добавил div класс
	.barcode {
	  font-family: "Libre Barcode 128 Text";
	  		  padding-top: 5px; 
	  font-size: 40px;
	  text-align: center;
	   page-break-after: always;
	}
```
ну и в самой форме обернул ` <div class="barcode"> {bar}  </div>`

**Reply by vldekov** [reply]
Супер! Большое спасибо.

**Reply by vldekov** [reply]
Почему то не получается вывести на печать штрихкод, Выводится само значение, а штрихкода нет. Что возможно я сделал не так?

**Reply by erpic-ru** [reply]
На сам компьютер нужно шрифт установить :)

**Reply by vldekov** [reply]
А где его можно взять? )

**Reply by erpic-ru** [reply]
https://fonts.google.com/specimen/Libre+Barcode+128

**Reply by vldekov** [reply]
Скачал, установил на ПК шрифт. Не работает. Мжет проблема в названии шрифта "font-family: "Libre Barcode 128 Text". У меня он называется "font-family: "Libre Barcode 128 обычный" или "LibreBarcode 128 Regular". У него несколько названий, а какой надо использовать?

**Reply by vldekov** [reply]
Заработало, спасибо!

**Reply by vldekov** [reply]
> <img alt="Снимок экрана 2023-01-23 в 09 58 50" width="285" src="https://user-images.githubusercontent.com/91674114/213982066-d9660175-ac59-4caf-9a8a-9ec2f8bd7322.png"> - вот 128
> 
> В печатной форме в стилях подключаете шрифт:
> 
> ```
> 
> @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600|Roboto:400,400i,700,700i,500|Roboto+Mono:400,700&amp;subset=cyrillic");
> body {font-family: 'Roboto', sans-serif; }
> 
> Ещё добавил div класс
> 	.barcode {
> 	  font-family: "Libre Barcode 128 Text";
> 	  		  padding-top: 5px; 
> 	  font-size: 40px;
> 	  text-align: center;
> 	   page-break-after: always;
> 	}
> ```
> 
> ну и в самой форме обернул ` <div class="barcode"> {bar} </div>`

Этот штрихкод не читается сканером (

**Comment by vldekov** [comment]
Ой! А полученный штрихкод не читается сканером. Куда копать?

**Reply by vldekov** [reply]
![Скриншот от 2023-01-24 08-03-49](https://user-images.githubusercontent.com/114401808/214210098-8db247d5-4306-4124-9233-dee2989ad3ca.png)

**Comment by vldekov** [comment]
https://github.com/lindell/JsBarcode/wiki/Browser-Setup
Может такой штрихкод попробовать использовать. Кто знает как это настроить в TOTUM?

**Comment by vldekov** [comment, marked as answer]
Может кому пригодится рабочее решение с печатью штрихкода code128 и др.
https://barcode.tec-it.com/barcode.ashx?data={DATA}&code=Code128&translate-esc=true

**Comment by vldekov** [comment]
![image](https://user-images.githubusercontent.com/114401808/214285365-0b1dfef4-c221-43b9-a433-674ec58b2b4e.png)
